vendor/github.com/containers/image/v5/docker/docker_transport.go

```diff
@@ -43,30 +43,44 @@ func (t dockerTransport) ValidatePolicyConfigurationScope(scope string) error {
 
 // dockerReference is an ImageReference for Docker images.
 type dockerReference struct {
-	ref reference.Named // By construction we know that !reference.IsNameOnly(ref)
+	ref             reference.Named // By construction we know that !reference.IsNameOnly(ref)
+	isUnknownDigest bool
 }
 
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an Docker ImageReference.
 func ParseReference(refString string) (types.ImageReference, error) {
 	if !strings.HasPrefix(refString, "//") {
 		return nil, fmt.Errorf("docker: image reference %s does not start with //", refString)
 	}
+	// Check if ref has UnknownDigestSuffix suffixed to it
+	unknownDigest := false
+	if strings.HasSuffix(refString, reference.UnknownDigestSuffix) {
+		unknownDigest = true
+		refString = strings.TrimSuffix(refString, reference.UnknownDigestSuffix)
+	}
 	ref, err := reference.ParseNormalizedNamed(strings.TrimPrefix(refString, "//"))
 	if err != nil {
 		return nil, err
 	}
+
 	ref = reference.TagNameOnly(ref)
-	return NewReference(ref)
+	return newReference(ref, unknownDigest)
 }
 
 // NewReference returns a Docker reference for a named reference. The reference must satisfy !reference.IsNameOnly().
 func NewReference(ref reference.Named) (types.ImageReference, error) {
-	return newReference(ref)
+	return newReference(ref, false)
+}
+
+// NewReferenceUnknownDigest returns a Docker reference for a named reference taking into account
+// the value of unknownDigest. The reference doesn't have to satisfy !reference.IsNameOnly()
+func NewReferenceUnknownDigest(ref reference.Named, unknownDigest bool) (types.ImageReference, error) {
```

Comment:
This new function might just hard-code `unknownDigest` to true instead of adding a parameter.

Callers with `false` can call the other API, and it seems unlikely to me that a generic caller would not know in which situation it is in. (And if it doesn’t know/care, it should accept a fully-formed `ImageReference` as a parameter from some other code that does know/care.)

Reply:
Fixed